NeoML/src/Dnn/Dnn.cpp

```diff
@@ -441,6 +442,17 @@ void CDnn::GetLayerList( CArray<const char*>& layerList ) const
 }
 
 CPtr<CBaseLayer> CDnn::GetLayer( const char* name )
+{
+	CBaseLayer* layer = getLayer( name );
+
+	NeoAssertMsg( !IsReferenceDnn()
+		|| dynamic_cast<CSourceLayer*>( layer ) != nullptr
+		|| dynamic_cast<CSinkLayer*>( layer ) != nullptr,
+		"For ReferenceDnn changing layers is restricted" );
```

Comment:
We are changing nothing here yet, this error message doesn't really help. Maybe suggest const version instead?
```suggestion
		"For ReferenceDnn changing layers is restricted. Use const version instead." );
```

Reply:
fixed